src/modules/song/thunks.ts

```diff
@@ -112,6 +112,21 @@ export const uploadSong = createAsyncThunk(
             },
           })
         );
+
+        if ("error" in generateArtistAgreementResponse) return;
+
+        if (
+          generateArtistAgreementResponse.meta.requestStatus === "fulfilled"
+        ) {
```

Comment:
I dont think this check is needed if `"error" in generateArtistAgreementResponse` is true, it will return and not run this code. Just adding it for now to make sure it only runs "fulfilled"